reactor-netty-http/src/test/java/reactor/netty/http/client/HttpClientProxyTest.java

```diff
@@ -203,6 +204,39 @@ public void shouldNotResolveTargetHostnameWhenMetricsDisabled() {
 				    .verify(Duration.ofSeconds(30));
 	}
 
+	@Test
+	public void shouldUseDifferentResolvers() {
+		HttpClient client =
+				HttpClient.create()
+				          .remoteAddress(server::address)
+				          .metrics(true, () -> MicrometerHttpClientMetricsRecorder.INSTANCE)
+				          .wiretap(true);
+
+		AtomicReference<AddressResolverGroup<?>> resolver1 = new AtomicReference<>();
+		client.doOnConnect(config -> resolver1.set(config.resolver()))
+		      .get()
+		      .uri("http://example.com")
+		      .responseSingle((response, body) -> Mono.zip(body.asString(),
+		                                                   Mono.just(response.responseHeaders())))
+		      .block(Duration.ofSeconds(30));
+
+		AtomicReference<AddressResolverGroup<?>> resolver2 = new AtomicReference<>();
+		client.proxy(ops -> ops.type(ProxyProvider.Proxy.HTTP)
+		                       .host("localhost")
+		                       .port(hoverflyRule.getProxyPort()))
+		      .doOnConnect(config -> resolver2.set(config.resolver()))
+		      .get()
+		      .uri(LOCALLY_NOT_RESOLVABLE_ADDRESS)
+		      .responseSingle((response, body) -> Mono.zip(body.asString(),
+		                                                   Mono.just(response.responseHeaders())))
+		      .block(Duration.ofSeconds(30));
+
```

Comment:
Should we add another http call here after the proxy just to verify that we are still able to make the calls without proxy and proxy configuration actually modified a copy?

```java
client.doOnConnect(config -> resolver1.set(config.resolver()))
		      .get()
		      .uri("http://example.com")
		      .responseSingle((response, body) -> Mono.zip(body.asString(),
		                                                   Mono.just(response.responseHeaders())))
		      .block(Duration.ofSeconds(30));
```

Reply:
yep let's extend the test